sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_client_manager.py

```diff
@@ -323,28 +334,81 @@ def __init__(
                 )
             )
             self._setup_failover_endpoints()
+            if load_balance:
+                random.shuffle(self._replica_clients)
             return
         raise ValueError("Please pass either endpoint and credential, or a connection string with a value.")
 
+    def get_next_client(self) -> Optional[_ConfigurationClientWrapper]:
+        """
+        Get the next client to be used for the request. find_active_clients needs be called before this method or None
+        client will be returned. If load_balance isn't enabled the first active client will be returned, which is the
+        provided endpoint unless it has failed. If load_balance is enabled the next client in the list will be returned.
+
+        :return: The next client to be used for the request.
+        """
+        if not self._active_clients:
+            self._last_active_client_name = ""
+            return None
+        if not self._load_balance:
+            return self._active_clients[0]
```

Comment:
So if `load_balance` is not enabled, we will always return the same client when we call this multiple times and we do not take note of the last active client. Is that the case? And if so, is that expected?

Reply:
Hm, now that you mention it, this breaks normal failover doesn't it, as the first one might be in a failed state.

Reply:
Fixed?

Reply:
I was thinking same. I would expect load-balancing to only introduce randomization in the order of the clients.

Reply:
I was thinking we would rather return the first active client after last used one.
 I am thinking of the situation where, for instance, we have 3 clients, we fail over from the first to the second client, and the second client has a long timeout such that by the time the call fails, the first client's backoff would have elapsed, this means we would failover back to the first client if we do not have load-balancing

Reply:
This is the non-load balancing code. We always use the same client, until it fails.